ledger/mary/mary.go

```diff
@@ -66,6 +66,15 @@ func (b *MaryBlock) UnmarshalCBOR(cborData []byte) error {
 	return nil
 }
 
+func (b *MaryBlock) MarshalCBOR() ([]byte, error) {
+	// Return the stored CBOR if available
+	if b.Cbor() != nil {
+		return b.Cbor(), nil
+	}
```

Comment:
This is cheating, and it won't work to give us proper CBOR in cases where we didn't start from CBOR

Reply:
It'll be nil, then and won't match.